ethtool: Introduce pause support

cathay4t opened the pull request:
Example:

```
  ethtool:
    pause:
      autoneg: false
      rx: true
      tx: true
```

Unit test cases included.
Integration test requires netdevsim kernel driver which does not exists
in CI environment. The test passes in Fedora 34 with
NetworkManager-1.31.4-28344.copr.0ce59b5dc4.fc34.x86_64

ffmancera commented on libnmstate/nm/ethtool.py:
```diff
@@ -0,0 +1,58 @@
+    if iface_ethtool.pause:
+        if iface_ethtool.pause.autoneg is not None:
+            nm_setting.option_set(
+                NM.ETHTOOL_OPTNAME_PAUSE_AUTONEG,
```
I think this is not correct. The linter complains because NetworkManager does not have that attribute.

ffmancera commented on libnmstate/nm/ethtool.py:
```diff
@@ -0,0 +1,58 @@
+            )
+        if iface_ethtool.pause.rx is not None:
+            nm_setting.option_set(
+                NM.ETHTOOL_OPTNAME_PAUSE_RX,
```
Same here.

ffmancera commented on libnmstate/nm/ethtool.py:
```diff
@@ -0,0 +1,58 @@
+            )
+        if iface_ethtool.pause.tx is not None:
+            nm_setting.option_set(
+                NM.ETHTOOL_OPTNAME_PAUSE_TX,
```
Same.

ffmancera reviewed (changes requested):
There some issues with pylint, please check them.

cathay4t committed:
ethtool: Introduce pause support

Example:

```
  ethtool:
    pause:
      autoneg: false
      rx: true
      tx: true
```

Unit test cases included.
Integration test requires netdevsim kernel driver which does not exists
in CI environment. The test passes in Fedora 34 with
NetworkManager-1.31.4-28344.copr.0ce59b5dc4.fc34.x86_64

Signed-off-by: Gris Ge <fge@redhat.com>

ffmancera reviewed (approved):
LGTM. Thanks!

cathay4t commented:
CI passed but github show it as hang. Force merging.
